### PR TITLE
Fix docs context-free commands subsection

### PR DESCRIPTION
**Pull request by cih9088**
### Summary
- Aligned docs
- Changed to `console` format

### Checklist

- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have added the description of my changes into [CHANGELOG](https://github.com/openvinotoolkit/datumaro/blob/develop/CHANGELOG.md).​
- [ ] I have updated the [documentation](https://github.com/openvinotoolkit/datumaro/tree/develop/docs) accordingly

### License

- [ ] I submit _my code changes_ under the same [MIT License](https://github.com/openvinotoolkit/datumaro/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2023 Intel Corporation
#
# SPDX-License-Identifier: MIT
```

### CODE REVIEW

File: datumaro/cli/commands/diff.py [hunk 1]
```diff
@@ -136,7 +136,7 @@ def _parse_comparison_method(s):
         type=float,
         help="IoU match threshold for shapes (default: %(default)s)",
     )
-    parser.add_argument(
+    distance_parser.add_argument(
```
Comment (vinnamkim) [inline datumaro/cli/commands/diff.py]:
```suggestion
    parser.add_argument(
```
I believe this should belong to `parser`.

**Review by vinnamkim — CHANGES_REQUESTED** [review]
I don't think that it is good idea to change `.py` when the release is super close. It would be good to leave documentation changes here and the other things should be targeted to `develop`.

File: datumaro/cli/commands/diff.py (hunk 1 above)
Comment (cih9088) [reply datumaro/cli/commands/diff.py]: No, this belongs to `distance` metric only. Please refer to [doc](https://github.com/openvinotoolkit/datumaro/blob/releases/1.2.0/docs/source/docs/command-reference/context_free/diff.md) and [code](https://github.com/openvinotoolkit/datumaro/blob/45d3f805eb2d44d433d03f6b1eb73249152a7975/datumaro/cli/commands/diff.py#L259).

Comment (cih9088) [reply datumaro/cli/commands/diff.py]: Anyway, reverted all python files changes.

File: docs/source/docs/command-reference/context_free/detect.md [hunk 2]
```diff
@@ -9,7 +9,7 @@ The detection result may be one of:
 
 - a single format being detected;
 - no formats being detected (if the dataset doesn't match any known format);
-- multiple formats being detected (if the dataset is ambiguous).
+- multiple formats being detected (if the dataset is ambiguous);
```
Comment (vinnamkim) [inline docs/source/docs/command-reference/context_free/detect.md]: `.` seems correct, is it?

File: docs/source/docs/command-reference/context_free/diff.md [hunk 3]
```diff
@@ -60,27 +60,37 @@ Parameters:
...
-`datum diff path/to/dataset1:voc path/to/dataset2:coco`
-
-- Compare the current working tree and a dataset:
-`datum diff path/to/dataset2:coco`
-
-- Compare a source from a previous revision and a dataset:
-`datum diff HEAD~2:source-2 path/to/dataset2:yolo`
+  and the `is_crowd` attribute from comparison
+  ```console
+  datum diff <path/to/other/project/>/ -if group -ia is_crowd
```
Comment (vinnamkim) [inline docs/source/docs/command-reference/context_free/diff.md]:
```suggestion
  datum diff <path/to/other/project/> -if group -ia is_crowd
```

File: docs/source/docs/command-reference/context_free/generate.md [hunk 4]
```diff
@@ -8,13 +8,13 @@ and saves it in the provided directory.
 Currently, can only generate fractal images, useful for network compression.
 To create 3-channel images, you should provide the number of images, height and width.
 The images are colorized with a model, which will be downloaded automatically.
-Uses the algorithm from the article: <https://arxiv.org/abs/2103.13023>
+Uses the algorithm from the [article](https://arxiv.org/abs/2103.13023)
```
Comment (vinnamkim) [inline docs/source/docs/command-reference/context_free/generate.md]:
```suggestion
Uses the algorithm from the [article](https://arxiv.org/abs/2103.13023).
```

File: docs/source/docs/command-reference/context_free/merge.md [hunk 5]
```diff
@@ -100,37 +92,59 @@ Parameters:
...
-
-Check groups of the merged dataset for consistency:
+- Merge 4 (partially-)intersecting projects,
+  - consider voting successful when there are no less than 3 same votes
+  - consider shapes intersecting when IoU >= 0.6
+  - check annotation groups to have `person`, `hand`, `head` and `foot`
+  (`?` is used for optional parts)
+
+  ```console
+  datum merge <path/to/project1/> <path/to/project2/> <path/to/project3/> <path/toproject4/> \
```
Comment (vinnamkim) [inline docs/source/docs/command-reference/context_free/merge.md]:
```suggestion
  datum merge <path/to/project1/> <path/to/project2/> <path/to/project3/> <path/to/project4/> \
```

File: docs/source/docs/command-reference/context_free/detect.md (hunk 2 above)
Comment (cih9088) [reply docs/source/docs/command-reference/context_free/detect.md]: Agreed. Fixed in https://github.com/openvinotoolkit/datumaro/pull/969/commits/bcc7a4dc3cda80689f191d08d7fe71991d34b05c

File: docs/source/docs/command-reference/context_free/merge.md (hunk 5 above)
Comment (cih9088) [reply docs/source/docs/command-reference/context_free/merge.md]: Fixed in https://github.com/openvinotoolkit/datumaro/pull/969/commits/bcc7a4dc3cda80689f191d08d7fe71991d34b05c

**Review by vinnamkim — APPROVED** [review]
LGTM.

**Review by sooahleex — APPROVED** [review]

**Pull request merged**